### ISSUE TITLE
Error 500 when `design/header/welcome` is null

### ISSUE THREAD

**Issue by gorghoa** [magento/magento2#37208]
### Preconditions and environment

- Magento version: 2.4.6 (upgrading from 2.4.5-p1)
- PHP 8.1

### Steps to reproduce

- The config `design/header/welcome` is null

### Expected result

- Nothing special, just not crashing.

### Actual result

Error 500, with exception when accessing the frontend:

`TypeError: addslashes(): Argument #1 ($string) must be of type string, null given in /var/www/html/app/vendor/magento/framework/Escaper.php:440`

<details>
  <summary>Log details</summary>

```
[2023-03-14T14:25:43.513719+00:00] main.CRITICAL: TypeError: addslashes(): Argument #1 ($string) must be of type string, null given in /var/www/html/app/vendor/magento/framework/Escaper.php:440
Stack trace:
#0 /var/www/html/app/vendor/magento/framework/Escaper.php(440): addslashes()
#1 /var/www/html/app/vendor/magento/module-theme/Block/Html/Header.php(62): Magento\Framework\Escaper->escapeQuote()
#2 /var/www/html/app/vendor/magento/module-theme/view/frontend/templates/html/header.phtml(11): Magento\Theme\Block\Html\Header->getWelcome()
#3 /var/www/html/app/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('...')
#4 /var/www/html/app/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php->render()
#5 /var/www/html/app/vendor/magento/framework/View/Element/Template.php(293): Magento\Framework\View\Element\Template->fetchView()
#6 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Framework\View\Element\Template->_toHtml()
#7 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#8 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#9 /var/www/html/app/vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#10 /var/www/html/app/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock()
#11 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#12 /var/www/html/app/vendor/magento/framework/View/Element/Html/Links.php(63): Magento\Framework\View\Layout->renderElement()
#13 /var/www/html/app/vendor/magento/framework/View/Element/Html/Links.php(83): Magento\Framework\View\Element\Html\Links->renderLink()
#14 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Framework\View\Element\Html\Links->_toHtml()
#15 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#16 /var/www/html/app/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#17 /var/www/html/app/vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#18 /var/www/html/app/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock()
#19 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#20 /var/www/html/app/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#21 /var/www/html/app/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#22 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#23 /var/www/html/app/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#24 /var/www/html/app/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#25 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#26 /var/www/html/app/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#27 /var/www/html/app/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#28 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#29 /var/www/html/app/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#30 /var/www/html/app/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#31 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#32 /var/www/html/app/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#33 /var/www/html/app/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#34 /var/www/html/app/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#35 /var/www/html/app/vendor/magento/framework/View/Layout.php(975): Magento\Framework\View\Layout->renderElement()
#36 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Layout->getOutput()
#37 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Layout\Interceptor->___callParent()
#38 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}()
#39 /var/www/html/app/generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout\Interceptor->___callPlugins()
#40 /var/www/html/app/vendor/magento/framework/View/Result/Page.php(260): Magento\Framework\View\Layout\Interceptor->getOutput()
#41 /var/www/html/app/vendor/magento/framework/View/Result/Layout.php(171): Magento\Framework\View\Result\Page->render()
#42 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Result\Layout->renderResult()
#43 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Result\Page\Interceptor->___callParent()
#44 /var/www/html/app/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}()
#45 /var/www/html/app/generated/code/Magento/Framework/View/Result/Page/Interceptor.php(32): Magento\Framework\View\Result\Page\Interceptor->___callPlugins()
#46 /var/www/html/app/vendor/magento/framework/App/Http.php(120): Magento\Framework\View\Result\Page\Interceptor->renderResult()
#47 /var/www/html/app/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#48 /var/www/html/app/pub/index.php(30): Magento\Framework\App\Bootstrap->run()
```

</details>

### WORKAROUND

If anyone stumbles upon the same issue : Set a value to the field in `Content` > `Configuration` > pick your scope > `Header welcome text` and you are good to go.

![image](https://user-images.githubusercontent.com/662768/225037800-0c18f6e6-7298-45fb-87bd-018017855913.png)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by gorghoa** [comment]
@magento give me 2.4-develop instance

**Comment by gorghoa** [comment]
https://github.com/magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @gorghoa ,

**_Issue confirmed !_**

Verified the issue in 2.4.6 Instance and 2.4-Develop instance.The issue is _**reproducible**_ in 2.4.6,But _**not**_ **_reproducible_** in 2.4-develop instance.

We can reproduce the issue in 2.4.6 Magento instance,Hence we are confirming the issue.

**_Preconditions:_**
Magento Version 2.4.6 and 2.4-develop
PHP Version 8.1

1._**The issue is not reproducible with 2.4-develop instance**_

**Kindly refer the below screenshots:**

<img width="1512" alt="fr3" src="https://user-images.githubusercontent.com/60198592/225288302-a154418e-94c8-43d0-90c1-5b652badb7da.png">

<img width="1512" alt="fr4" src="https://user-images.githubusercontent.com/60198592/225288349-338e1238-81cb-443a-85fe-a3767863b867.png">

_**Steps to reproduce:**_

- Install Magento instance 2.4 - develop
- Go to Admin Panel and click the _**content**_ option
- Select **_Configuration_** option under **_Design_**
- Click the edit button of Main website store view 
- Click the header  option and Select the Welcome text box
- Delete the _**default welcome message**_ and save the configuration
- Go to front end hit the url.

In 2.4-develop instance when we are accessing the front end url with default welcome Message as null,its working without any issue,Without welcome msg front end running properly.So the issue is not reproducible in 2.4-develop.

2.**_But The issue is reproducible in 2.4.6_**

**Kindly refer the below screenshots:**

<img width="1512" alt="fr2" src="https://user-images.githubusercontent.com/60198592/225284569-992efe46-3da4-4727-90b5-a638883b04a7.png">

<img width="1512" alt="fr1" src="https://user-images.githubusercontent.com/60198592/225284491-a0081410-3181-447a-ace3-6b3b2cfd36d5.png">

**_Steps to reproduce:_**

- Install Magento instance 2.4.6
- Go to Admin Panel and click the _**content**_ option
- Select **_Configuration_** option under **_Design_**
- Click the edit button of Main website store view 
- Click the header  option and Select the Welcome text box
- Delete the _**default welcome message**_ and save the configuration
- Go to front end hit the url.


When we accessing the front with default welcome message as null,Its showing error with exception as per actual result and this is reproducible in 2.4.6 magento instance,Hence we are confirming the issue.

Regards,

**Comment by slavvka** [comment]
@magento give me 2.4-develop instance

**Comment by Viper9x** [comment]
@magento I am working on this

**Comment by chrisbadley** [comment]
> @magento I am working on this

I used your commit earlier to apply the fix, which worked well. Thanks.

**Comment by Viper9x** [comment]
I've created a pull request to solve this issue and it was approved yesterday.
You can apply this patch to your project:
```
From 5223d7e9c911f3a19a7ed42b28127b2a1afe7e1d Mon Sep 17 00:00:00 2001
From: Tu Van <vandinhtuit@gmail.com>
Date: Thu, 16 Mar 2023 14:20:38 +0700
Subject: [PATCH] Fix PHP Type error in Header block class

---
 vendor/magento/module-theme/Block/Html/Header.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vendor/magento/module-theme/Block/Html/Header.php b/vendor/magento/module-theme/Block/Html/Header.php
index 1550ebaa367d..e7eafc0b7316 100644
--- a/vendor/magento/module-theme/Block/Html/Header.php
+++ b/vendor/magento/module-theme/Block/Html/Header.php
@@ -54,7 +54,7 @@ public function __construct(
     public function getWelcome()
     {
         if (empty($this->_data['welcome'])) {
-            $this->_data['welcome'] = $this->_scopeConfig->getValue(
+            $this->_data['welcome'] = (string)$this->_scopeConfig->getValue(
                 'design/header/welcome',
                 ScopeInterface::SCOPE_STORE
             );

``` 
Document on how to apply a composer patch:
- https://experienceleague.adobe.com/docs/commerce-operations/upgrade-guide/patches/apply.html
- https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/how-to/how-to-apply-a-composer-patch-provided-by-magento.html?lang=en

Pull request: #37222

**Comment by gorghoa** [comment]
Thank you to everyone involved :)

**Comment by chittima** [comment]
Issue is closed as it is not reproducible on latest 2.4-develop

**Comment by hostep** [comment]
After a git bisect session, I've found the fix in core magento for this: https://github.com/magento/magento2/commit/4b5c1dc1382c2e6e144789c49db27c9085146551

@Viper9x: it looks like [your PR](https://github.com/magento/magento2/pull/37222) is no longer necessary.

**Comment by Viper9x** [comment]
Good catch @hostep . Thanks.

Although the purpose of that commit (https://github.com/magento/magento2/commit/4b5c1dc1382c2e6e144789c49db27c9085146551) was to fix the Welcome text is untranslatable, it also inadvertently solved this issue since `__()` method return `\Magento\Framework\Phrase` which also [force change the type of the text to string ](https://github.com/magento/magento2/blob/2.4.6/lib/internal/Magento/Framework/Phrase.php#L72)

**Comment by lano-vargas** [comment]
This appears to be still an issue for me when I've upgrade it to 2.4.6-p4

**Comment by hostep** [comment]
That's correct, the fix will be included in 2.4.7
